sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotDecreaseMemberVisibility.cs

```diff
@@ -166,13 +174,23 @@ private static bool IsMatchingSignature(IMethodSymbol baseMethod, IMethodSymbol
 
             private static bool AreParameterTypesEqual(IParameterSymbol p1, IParameterSymbol p2)
             {
+                if (p1.RefKind != p2.RefKind)
+                {
+                    return false;
+                }
+
                 return p1.Type.TypeKind == TypeKind.TypeParameter ?
                          p2.Type.TypeKind == TypeKind.TypeParameter
                          : Equals(p1.Type.OriginalDefinition, p2.Type.OriginalDefinition);
             }
 
-            private static bool IsDecreasingAccess(Accessibility baseAccess, Accessibility memberAccess)
+            private static bool IsDecreasingAccess(Accessibility baseAccess, Accessibility memberAccess, bool isOverride)
             {
+                if (memberAccess == Accessibility.NotApplicable && isOverride)
```

Comment:
This is a bit weird. It is fine to omit get or set when overriding, but when present, the access modifier needs to match the base.

Will add a test if there isn't one already.